File: IGListTest/IGListTest/Views/JournalEntryCell.m
//
//  JournalEntryCell.m
//  IGListTest
//
//  Created by liuruixuan on 2023/4/16.
//

#import "JournalEntryCell.h"
#import "Utils.h"

#define inset UIEdgeInsetsMake(0, 15, 0, 15)

@interface JournalEntryCell ()



@end

@implementation JournalEntryCell

- (instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        self.contentView.backgroundColor = ssRGBHex(0x0c1f3f);
        [self.contentView addSubview:self.label];
    }
    return self;
}

- (void)layoutSubviews {
    [super layoutSubviews];
    self.label.frame = UIEdgeInsetsInsetRect(self.bounds, inset);
}

+ (CGSize)cellSizeWithWidth:(CGFloat)width text:(NSString *)text {
    return [Utils sizeWithText:text font:AppFontDefault width:width insets:inset].size;
}

- (UILabel *)label {
    if (!_label) {
        _label = [[UILabel alloc] init];
        _label.font = AppFontDefault;
        _label.numberOfLines = 0;
        _label.textColor = [UIColor whiteColor];
        _label.backgroundColor = [UIColor clearColor];
    }
    return _label;
}

@end

File: Marslink_Swift5/Marslink_Swift5/Views/JournalEntryCell.swift
//
//  JournalEntryCell.swift
//  Marslink_Swift5
//
//  Created by liuruixuan on 2023/4/17.
//

import UIKit

class JournalEntryCell: UICollectionViewCell {
    static let font = AppFont()
    static let inset = UIEdgeInsets(top: 0, left: 15, bottom: 0, right: 15)

    static func cellSize(width: CGFloat, text: String) -> CGSize {
        return TextSize.size(text, font: JournalEntryCell.font, width: width, insets: JournalEntryCell.inset).size
    }

    let label: UILabel = {
        let label = UILabel()
        label.backgroundColor = UIColor.clear
        label.numberOfLines = 0
        label.font = JournalEntryCell.font
        label.textColor = UIColor.white
        return label
    }()

    override init(frame: CGRect) {
        super.init(frame: frame)
        contentView.backgroundColor = UIColor(hex6: 0x0C1F3F)
        contentView.addSubview(label)
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    override func layoutSubviews() {
        super.layoutSubviews()
        label.frame = bounds.inset(by: JournalEntryCell.inset)
    }
}
